api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirObservationDaoImpl.java

```diff
@@ -59,6 +63,10 @@ protected void setupSearchParams(Criteria criteria, SearchParameterMap theParams
 					entry.getValue()
 					        .forEach(category -> handleConceptClass(criteria, (TokenAndListParam) category.getParam()));
 					break;
+				case FhirConstants.BASED_ON_REFERENCE_SEARCH_HANDLER:
```

Comment:
So, for `basedOnReferences` we don't want to add *that* to the Hibernate search, because Hibernate won't know what that means at all. What we need to do is tell Hibernate to join the relevant table. For `basedOn`, that would be the orders table. So, what we actually want to do here is ensure that we call something like:

```java
if (lacksCriteria(criteria, "ord") {
    criteria.createAlias("order", "ord");
}

// handle any properties of the order here
```